### PR TITLE
replace axios with native fetch

### PR DESCRIPTION
**Pull request by cjpearson**
WIP replacement of axios

Breaking changes: 

- axiosAdapter support removed
- ~~no basic auth~~
- ~~fetch does not throw errors on non-200 responses~~

### CODE REVIEW

File: src/helpers/BapiClient.ts [hunk 1]
```diff
@@ -730,18 +726,9 @@ export class BapiClient {
...
-        );
-        return response;
-      } catch (e) {
-        if (axios.isAxiosError(e) && e.response?.status === 404) {
-          return undefined;
-        }
-
-        throw e;
-      }
+      return await this.execute(
```
Comment (HenriBeck) [inline src/helpers/BapiClient.ts]: TODO: Check this to return the correct error

File: src/helpers/execute.ts [hunk 2]
```diff
@@ -67,43 +65,28 @@ export async function execute<SuccessResponseT>(
...
+      ...(auth && auth.type === 'token' ? {'X-Access-Token': auth.token} : {}),
     },
-    url,
     method: bapiCall.method,
-    data:
+    body:
       bapiCall.method === 'POST' || bapiCall.method === 'PATCH'
         ? bapiCall.data
         : undefined,
-    validateStatus: acceptAllResponseCodes
```
Comment (HenriBeck) [inline src/helpers/execute.ts:77]: The validate status logic is missing right?

File: src/helpers/BapiClient.ts (hunk 1 above)
Comment (cjpearson) [reply src/helpers/BapiClient.ts]: Mostly reverted so it should have the same behavior on 404 now

File: src/helpers/execute.ts (hunk 2 above)
Comment (cjpearson) [reply src/helpers/execute.ts:77]: Oops. Added that back

**Review by HenriBeck — APPROVED** [review]

File: src/helpers/execute.ts [hunk 3]
```diff
@@ -67,43 +66,33 @@ export async function execute<SuccessResponseT>(
...
+      ...(auth && auth.type === 'token' ? {'X-Access-Token': auth.token} : {}),
+      ...(auth && auth.type === 'basic' ? { 'Authorization': 'Basic ' + btoa(auth.username + ":" + auth.password) } : {})
     },
-    url,
     method: bapiCall.method,
-    data:
+    body:
       bapiCall.method === 'POST' || bapiCall.method === 'PATCH'
-        ? bapiCall.data
+        ? bapiCall.data ? JSON.stringify(bapiCall.data) : undefined
```
Comment (HenriBeck) [inline src/helpers/execute.ts]:
```suggestion
        ? JSON.stringify(bapiCall.data)
```

File: src/helpers/execute.ts [hunk 4]
```diff
@@ -67,43 +66,33 @@ export async function execute<SuccessResponseT>(
...
-  if (auth) {
-    if (auth.type === 'token') {
-      fetchOptions.headers!['X-Access-Token'] = auth.token;
-    } else {
-      fetchOptions.auth = {
-        username: auth.username,
-        password: auth.password,
-      };
-    }
+  if (!response.ok && !acceptAllResponseCodes) {
```
Comment (HenriBeck) [inline src/helpers/execute.ts:79]: 👍

**Review by HenriBeck — APPROVED** [review]

**Pull request merged**